### ISSUE TITLE
Performance Improvement?

### ISSUE THREAD

**Issue by DriftWare07** [Zylann/godot_heightmap_plugin#344]
As is, the actual terrain in the editor makes it very laggy. Are there any performance tips to help reduce this?

**Comment by Calinou** [comment]
Which hardware do you have? Which Godot version are you using? If using a self-compiled build, did you build it with `production=yes` SCons option?

**Comment by DriftWare07** [comment]
I'm using standard Godot 3.5. as for gpu, I'm using an rx 50. Sorry about
the late reply.

**Comment by Calinou** [comment]
>  as for gpu, I'm using an rx 50.

What's an RX 50? I've never heard of this GPU model.

**Comment by DriftWare07** [comment]
Sorry, rx 570.

**Comment by Zylann** [comment]
An RX 570: https://www.amd.com/en/products/graphics/radeon-rx-570
Compared with an nVidia 1060: https://www.nvidia.com/en-gb/geforce/graphics-cards/geforce-gtx-1060/specifications/
Based on clock speeds, RX 570 is about 20% slower than a 1060.

As for lag, I have a 1060, and dont have lag at all. Can you provide a test project, tell us how you are measuring performance in it, and how much you get?

Things that can lower performance are having a lot of detail layer nodes at high distance, or have low chunk size with high LOD distance.

**Comment by DriftWare07** [comment]
I'm out of town as of now, but I can get back to you asap.

**Comment by mojoyup** [comment]
I am going to watch this also because I suddenly started having issues when loading the "map" during runtime whereas it takes about 20-21 seconds to load where before it was within seconds. I testing taking everything in/out of the scene and narrowed it down to this plugin. Just chiming in to give insight on performance issues as well and hopefully help with conclusions.

**Comment by Calinou** [comment]
> I am going to watch this also because I suddenly started having issues when loading the "map" during runtime whereas it takes about 20-21 seconds to load where before it was within seconds. I testing taking everything in/out of the scene and narrowed it down to this plugin. Just chiming in to give insight on performance issues as well and hopefully help with conclusions.

Did you save binary data into text-based scene/resource files, causing them to contain Base64 serialized data (which is slow to read/write)?

**Comment by mojoyup** [comment]
> > I am going to watch this also because I suddenly started having issues when loading the "map" during runtime whereas it takes about 20-21 seconds to load where before it was within seconds. I testing taking everything in/out of the scene and narrowed it down to this plugin. Just chiming in to give insight on performance issues as well and hopefully help with conclusions.
> 
> Did you save binary data into text-based scene/resource files, causing them to contain Base64 serialized data (which is slow to read/write)? You need to save it to external binary `.res` files.

It is automatic, so it may be likely that the data (within the referenced folder of choice) is not a .res. I can check into that. And external as in via save/load from user rather than root right?

**Comment by Calinou** [comment]
> And external as in via save/load from user rather than root right?

No, it means saving to resources that are not built into the `.tscn` filer;

**Comment by mojoyup** [comment]
> > And external as in via save/load from user rather than root right?
> 
> No, I mean saving resources such as Mesh to files that are not built into the `.tscn` file.

Oh ok, no that has not been done. All of the data is strictly from the default saving/loading of the plugin still.
